js/main.js

```diff
@@ -0,0 +1,49 @@
+const checkCorrectNumber = (value) => {
+  if (Number.isNaN(value) === true || typeof value !== 'number' || Number.isFinite(value) !== true) {
+    throw new RangeError('Числа должны быть положительными или 0.');
+  }
+
+  if (value < 0) {
+    value = 0;
+  } else if (value > Number.MAX_SAFE_INTEGER) {
+    value = Number.MAX_SAFE_INTEGER;
+  }
+
+  return value;
+};
```

Comment:
Методы с приставкой `is*` возвращают булевое значение, поэтому нам не нужно сравнивать их с булевыми значениями еще раз.
Не нужно мутировать (изменять) параметры функции `value = 0`.
Посмотри описание метода [isFinite](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite), он может уже проверять некторые случаи, которые здесь проверяем сами.
```suggestion
const checkCorrectNumber = (value) => {
  if (!Number.isFinite(value)) {
    throw new RangeError('Числа должны быть положительными или 0.');
  }

  if (value < 0) {
    return 0;
  }
  
  if (value > Number.MAX_SAFE_INTEGER) {
    return Number.MAX_SAFE_INTEGER;
  }
};
```

Reply:
Убрал лишние проверки, оставил только isFinite + убрал мутации для value